### PR TITLE
Refactored RepetitionQubit

### LINKED ISSUE #60: [RepCode] Extend RepetitionQubit from XXZZQubit
Extend `RepetitionQubit` from `XXZZQubit` (or at least `RotatedQubit`), as the Repetition code can be thought of as a very slim surface code. This would follow the work from #27 and #28.

Add a `RepetitionPhaseQubit` as well as a "row" repetition code.

### PR DESCRIPTION
**Pull request by amirebrahimi**
Resolves #60 

Needs simulator fixes from @Phionx for the `2-fitters-lookup.ipynb` and `3-benchmarking.ipynb` notebooks to work.

### CODE REVIEW

File: qtcodes/fitters/lattice_decoder.py [hunk 1]
```diff
@@ -425,6 +425,7 @@ def draw(
         dpi: Optional[int] = None,
         node_size: Optional[int] = None,
         font_size: Optional[float] = None,
+        show: Optional[bool] = True,
```
Comment (Phionx) [inline qtcodes/fitters/lattice_decoder.py:428]: Please add to the doc string to reflect this change.

File: qtcodes/fitters/lattice_decoder.py [hunk 2]
```diff
@@ -494,7 +497,8 @@ def node_to_pos3D(node):
         # 3D network plot
         with plt.style.context(("ggplot")):
             fig = plt.figure(figsize=(20, 14))
-            ax = Axes3D(fig)
+            ax = Axes3D(fig, auto_add_to_figure=False)
```
Comment (Phionx) [inline qtcodes/fitters/lattice_decoder.py:500]: Why is this change necessary? (Is the original code being deprecated?)

File: qtcodes/fitters/rotated_surface.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def _make_syndrome_graph(self) -> None:
         Returns:
         """
         start_nodes = {"Z": (0.5, 0.5), "X": (0.5, 1.5)}
-        for syndrome_graph_key in ["X", "Z"]:
+        for syndrome_graph_key in self.syndrome_graph_keys:
```
Comment (Phionx) [inline qtcodes/fitters/rotated_surface.py:59]: Good catch!

File: tests/rep.py [hunk 4]
```diff
@@ -71,11 +75,7 @@ def test_single_errors_rep(self):
 
                 # Simulate
                 results = (
-                    execute(
-                        qubit.circ,
-                        Aer.get_backend("aer_simulator"),
-                        shots=1000,
-                    )
+                    execute(qubit.circ, Aer.get_backend("aer_simulator"), shots=1000,)
```
Comment (Phionx) [inline tests/rep.py:78]: nit. Just make sure this is `black` formatted.

File: qtcodes/fitters/repetition.py [hunk 5]
```diff
@@ -20,88 +17,14 @@ class RepetitionDecoder(LatticeDecoder):
...
+        if "d" not in self.params:
+            self.params["d"] = (3, 1)
+        elif isinstance(self.params["d"], Number):
+            d = int(self.params["d"])
+            self.params["d"] = (d, 1)
         else:
-            raise NotImplementedError(
-                "Only Z readout is supported in the Repetition code."
-            )
+            self.params["d"] = (self.params["d"][0], 1)
```
Comment (Phionx) [inline qtcodes/fitters/repetition.py:28]: This allows the user to pass in `"d": (3,2)` and still end up with `self.params["d"] = (3,1)`. Let's make sure that `self.params["d"][1] == 1` and otherwise raise an exception. Also, let's use the constants you made, instead of `[0]` and `[1]`. 

We were briefly chatting about a (phase-flip-protected) repetition qubit protected against Z flips. On a related note, we could allow users to pass in `"d": (1,3), "phase-flip-protected": True` to `RepetitionQubit` when they want to create a  phase-flip-protected repetition qubit. And, in the case where a user just sends in `"d": (1,3)` without setting `"phase-flip-protected": True`, we can raise a warning stating that the user is creating a phase-flip-protected repetition qubit instead of a possibly desired bit-flip-protected repetition qubit. 

To enable this phase-flip-protected functionality (tracked by issue #66), I believe we will just need to modify this parameter validation and add some tests. However, we can leave that for a separate PR, but feel free to implement the phase-flip-protected functionality in this PR.

File: qtcodes/circuits/repetition.py [hunk 6]
```diff
@@ -1,438 +1,37 @@
...
+        params = params if params else {}
+
+        # The repetition code is a 1D lattice
+        if "d" not in params:
+            params["d"] = (3, 1)
+        elif isinstance(params["d"], Number):
+            d = int(params["d"])
+            params["d"] = (d, 1)
+        else:
+            params["d"] = (params["d"][0], 1)
```
Comment (Phionx) [inline qtcodes/circuits/repetition.py:35]: You should use the constants you made here as well, instead of `["d"][0]`. 

Also, all my comments in `fitters/repetition.py`  parameter validation apply here as well.

**Review by Phionx — CHANGES_REQUESTED** [review]
I've added some minor comments, and created issue #66 to track work towards implementing phase-flip-protected repetition code qubits.

File: qtcodes/fitters/lattice_decoder.py (hunk 2 above)
Comment (amirebrahimi) [reply qtcodes/fitters/lattice_decoder.py:500]: Yes.

File: qtcodes/fitters/repetition.py (hunk 5 above)
Comment (amirebrahimi) [reply qtcodes/fitters/repetition.py]: This is fixed now

File: qtcodes/fitters/lattice_decoder.py (hunk 1 above)
Comment (amirebrahimi) [reply qtcodes/fitters/lattice_decoder.py:428]: Fixed

File: qtcodes/circuits/repetition.py (hunk 6 above)
Comment (amirebrahimi) [reply qtcodes/circuits/repetition.py]: Fixed

**Comment by amirebrahimi** [comment]
@Phionx - changes made; please review

**Comment by Phionx** [comment]
Closing this PR, in favor of #69 , which encompasses these changes/commits.
